packages/mongo/mongo_driver.js

```diff
@@ -180,8 +180,10 @@ MongoConnection = function (url, options) {
         }
 
         // First, figure out what the current primary is, if any.
-        if (db.serverConfig._state.master)
-          self._primary = db.serverConfig._state.master.name;
+        if (db.serverConfig.isMasterDoc) {
+          self._primary = db.serverConfig.isMasterDoc.primary;
```

Comment:
`isMasterDoc` seems like a strange name for something other than a boolean value, but I believe you if you're sure this works.